engine/src/main/java/org/hibernate/validator/constraints/time/PeriodMax.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.constraints.time;
+
+import static java.lang.annotation.ElementType.ANNOTATION_TYPE;
+import static java.lang.annotation.ElementType.CONSTRUCTOR;
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.PARAMETER;
+import static java.lang.annotation.ElementType.TYPE_USE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.Repeatable;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+import java.time.Period;
+import javax.validation.Constraint;
+import javax.validation.Payload;
+import javax.validation.ReportAsSingleViolation;
+
+import org.hibernate.validator.constraints.time.PeriodMax.List;
+
+/**
+ * Annotated {@link Period} element's length must be less than or equal to
+ * the one constructed from {@link PeriodMax#years()}, {@link PeriodMax#months()},
+ * {@link PeriodMax#days()} values.
+ *
+ * @author Marko Bekhta
+ * @hv.experimental
+ */
+@Documented
+@Constraint(validatedBy = { })
+@Target({ METHOD, FIELD, ANNOTATION_TYPE, CONSTRUCTOR, PARAMETER, TYPE_USE })
+@Retention(RUNTIME)
+@Repeatable(List.class)
+@ReportAsSingleViolation
+public @interface PeriodMax {
+
+	String message() default "{org.hibernate.validator.constraints.time.PeriodMax.message}";
+
+	Class<?>[] groups() default { };
+
+	Class<? extends Payload>[] payload() default { };
+
+	int years() default 0;
+
+	int months() default 0;
+
+	int days() default 0;
+
+	/**
```

Comment:
> I'm wondering if it really makes sense to consider Period baring these limitations.

Yes, I'm getting doubts as well.

I first thought one could construct something around `normalize()`,  but after all you cannot really tell whether "2 months" is longer or shorter than "60 days", as you don't know the number of days in a months. So it seems for a good reason that the java.time API doesn't expose a way for comparing periods. We then shouldn't try to invent it.

+1 for supporting `Duration` only.